### PR TITLE
feat: adds support for disableMigration on repository mixin

### PR DESCRIPTION
**Pull request by marioestradarosa**
Signed-off-by: Mario Estrada <marioestradarosa@yahoo.com>

Developers have been asking a way to make a **DataSource** _readonly_ when migrating the database schema on certain environments. For a non-breaking change the property ```disableMigration``` is introduced that can be specified in the **DataSource** _configuration_, which behaves in the way it does currently if not specified or if set to false, so only when set to true it will skip that datasource when running the migrate script or when auto-migrate.

It can work together with the ```process.env.NODE_ENV``` in order to seamlessly achieve the readonly mode (not migrate flag).  The reason why I did not followed the pattern of models: property in the ```migrateSchemaoptions``` is because it is really easier to work along with the ```process.env.NODE_ENV``` for these use cases with the property flagged in the configuration.

## Checklist

- [ ] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Review by dhmlau — APPROVED** [review]
Thanks @marioestradarosa for the PR. The changes LGTM. Please get at least one more approval before landing. Thanks!

File: docs/site/Database-migrations.md
Comment (raymondfeng) [inline docs/site/Database-migrations.md:155]: Maybe `disableMigration` is better with default value to `false`.

Comment (marioestradarosa) [reply docs/site/Database-migrations.md]: > Maybe `disableMigration` is better with default value to `false`.

Agree. @raymondfeng  PTAL.

**Review by raymondfeng — APPROVED** [review]

**Pull request merged**
